hydroflow/tests/surface_async.rs

```diff
@@ -263,3 +264,106 @@ pub async fn test_futures_stream_sink() -> Result<(), Box<dyn Error>> {
 
     Ok(())
 }
+
+#[tokio::test(flavor = "current_thread")]
+async fn asynctest_dest_sink_bounded_channel() {
+    // This is for testing. Don't actually use unbounded channels, as their input will end up
+    // buffered regardless. Instead use `-> for_each(|x| { send.send(x).unwrap(); })` with an
+    // unbounded channel sender `send`.
+    let (send, recv) = tokio::sync::mpsc::channel::<usize>(5);
+    let send = tokio_util::sync::PollSender::new(send);
+    let mut recv = tokio_stream::wrappers::ReceiverStream::new(recv);
+
+    let mut flow = hydroflow_syntax! {
+        source_iter(0..10) -> dest_sink(send);
+    };
+    tokio::time::timeout(std::time::Duration::from_secs(1), flow.run_async())
+        .await
+        .expect_err("Expected time out");
+
+    // Only 5 elemts received due to buffer size
+    let out: Vec<_> = recv_into(&mut recv);
+    assert_eq!(&[0, 1, 2, 3, 4], &*out);
+}
+
+#[tokio::test(flavor = "current_thread")]
```

Comment:
Now I'm real confused. I thought `dest_asyncwrite` is for byte streams and `dest_sink` is for collections of discrete objects.  Which one are you testing here?